src/backend/vault-common/aes-gcm.c

```diff
@@ -25,8 +25,7 @@
 #define AES_PAD_NONE CX_PAD_NONE
 #define AES_CHAIN_ECB CX_CHAIN_ECB
 
-#define memcpy os_memcpy
-#define memset os_memset
+#define memcpy memmove
```

Comment:
Why memcpy is defined on memove?

Reply:
`os_memcpy` and `os_memset` are deprecated in the SDK so it triggers warning.
Nevertheless the `os_memcpy` declaration in the SDK call `memmove` that's why I define `memcpy` as `memmove` to mimic the old behaviour.
Maybe should I add a comment on this?

Reply:
I think that the define should be removed completely because it is confusing to have a memcpy that performs a memmove. To me in this code, using memcpy is correct as there is no reason that the two memory blocks overlap

Reply:
Agree